pumps/hybrid.go

```diff
@@ -62,121 +111,145 @@ func (p *HybridPump) New() Pump {
 }
 
 func (p *HybridPump) Init(config interface{}) error {
-
 	p.log = log.WithField("prefix", hybridPrefix)
 
-	meta := config.(map[string]interface{})
-	// read configuration
-	rpcConfig := rpc.Config{}
-	if useSSL, ok := meta["use_ssl"]; ok {
-		rpcConfig.UseSSL = useSSL.(bool)
-	}
-	if sslInsecure, ok := meta["ssl_insecure_skip_verify"]; ok {
-		rpcConfig.SSLInsecureSkipVerify = sslInsecure.(bool)
-	}
-	if connStr, ok := meta["connection_string"]; ok {
-		rpcConfig.ConnectionString = connStr.(string)
+	// Read configuration file
+	p.hybridConfig = &HybridPumpConf{}
+	err := mapstructure.Decode(config, &p.hybridConfig)
+	if err != nil {
+		p.log.Error("Failed to decode configuration: ", err)
+		return err
 	}
-	if rpcKey, ok := meta["rpc_key"]; ok {
-		rpcConfig.RPCKey = rpcKey.(string)
+
+	processPumpEnvVars(p, p.log, p.hybridConfig, hybridDefaultENV)
+
+	if p.hybridConfig.ConnectionString == "" {
+		p.log.Error("Failed to decode configuration - no connection_string")
+		return errors.New("empty connection_string")
 	}
-	if apiKey, ok := meta["api_key"]; ok {
-		rpcConfig.APIKey = apiKey.(string)
+
+	p.hybridConfig.CheckDefaults()
+
+	p.log.Info("connecting to MDCB rpc server")
+	errConnect := p.connectRPC()
+	if errConnect != nil {
+		p.log.Fatal("Failed to connect to RPC server")
 	}
-	if groupID, ok := meta["group_id"]; ok {
-		rpcConfig.GroupID = groupID.(string)
+	p.log.Info("starting rpc dispatcher")
+	p.startDispatcher()
+
+	p.log.Info("loging in to MDCB rpc server")
+	logged, err := p.RPCLogin()
```

Comment:
what would happen if the connection get lost after this? what if we continue checking that we can Login to the server, and if it fails then we attempt to reconnect?